src/eval.ts

```diff
@@ -0,0 +1,92 @@
+import { Flag, Shard, Range, Variation } from './interfaces';
+import { matchesRule } from './rules';
+import { Sharder } from './sharders';
+
+export interface FlagEvaluation {
+  flagKey: string;
+  subjectKey: string;
+  subjectAttributes: Record<string, string | number | boolean>;
+  allocationKey: string | null;
+  variation: Variation | null;
+  extraLogging: Record<string, string>;
+  doLog: boolean;
+}
+
+export class Evaluator {
+  sharder: Sharder; // Assuming a Sharder type exists, replace 'any' with 'Sharder' when available
```

Comment:
Is this a comment leftover from development? I don't understand it

Reply:
Good catch -- auto-generated when translating the Python code. Removing